wasm/wasm.go

```diff
@@ -109,3 +122,27 @@ type Memory interface {
 	// Write writes the slice to the underlying buffer at the offset or returns false if out of range.
 	Write(offset uint32, v []byte) bool
 }
+
+// EncodeF32 converts the input so that it can be used as a Function F32 parameter or result.
+// See DecodeF32
+func EncodeF32(input float32) uint64 {
+	return uint64(math.Float32bits(input))
+}
+
+// DecodeF32 converts the Function F32 parameter or result to a float32.
+// See DecodeF32
+func DecodeF32(input uint64) float32 {
+	return float32(math.Float64frombits(input))
```

Comment:
not sure, but I think this should be?

```suggestion
	return math.Float32frombits(uint32(input))
```

Reply:
didn't work in signed case! I owe backfilling of this test :D